execute/tokendata/observer.go

```diff
@@ -116,18 +116,41 @@ func (c *compositeTokenDataObserver) Observe(
 	msgObservations exectypes.MessageObservations,
 ) (exectypes.TokenDataObservations, error) {
 	tokenDataObservations := c.initTokenDataObservations(msgObservations)
+	// Terminate immediately if there are no observers
+	if len(c.observers) == 0 {
+		return tokenDataObservations, nil
+	}
+
+	resultsCh := make(chan exectypes.TokenDataObservations, len(c.observers))
+	defer close(resultsCh)
 
 	for _, ob := range c.observers {
-		tokenData, err := ob.Observe(ctx, msgObservations)
-		if err != nil {
-			c.lggr.Error("Error while observing token data", "error", err)
-			continue
-		}
-		tokenDataObservations, err = merge(tokenDataObservations, tokenData)
-		if err != nil {
-			return nil, err
+		go func(ob *TokenDataObserver) {
+			tokenData, err := (*ob).Observe(ctx, msgObservations)
+			// Best effort, if one observer fails we still want to process the rest of tokens
+			if err != nil {
+				c.lggr.Errorw("Error while observing token data", "error", err)
+				tokenData = exectypes.TokenDataObservations{}
+			}
+			resultsCh <- tokenData
+		}(&ob)
+	}
+
+	for i := 0; i < len(c.observers); i++ {
+		select {
```

Comment:
The granularity of a result is an entire set of observations which causes problems when the context is done:
1. The observers are notified to start returning accumulated attestations.
2. The `ctx.Done()` case is hit when immediately returns accumulated attestations.

This means we're very likely to ignore all attestations when the timeout is hit.